components/engine/src/iterable.rs

```diff
@@ -128,32 +128,35 @@ pub trait Iterable {
     }
 
     // Seek the first key >= given key, if not found, return None.
+    // TODO: Make it zero-copy.
```

Comment:
how can we avoid this? @breeswish

Reply:
seems we can just return `Result<Option<&[u8], &[u8]>>`

Reply:
Yes, we can do it later in an individual PR.